apps/pipeline/lib/pipeline/writer/s3_writer.ex

```diff
@@ -96,6 +97,17 @@ defmodule Pipeline.Writer.S3Writer do
     end
   end
 
+  @impl Pipeline.Writer
+  @spec delete(dataset: [term()]) :: :ok | {:error, term()}
+  def delete(args) do
+    table_name =
+      Keyword.fetch!(args, :dataset)
+      |> StatementUtils.parse_table_name()
+
+    delete_orc_table(table_name)
+    delete_json_table(table_name)
+  end
+
```

Comment:
Does the S3Writer need to implement this callback?  Or does the TableWriter just take care of deleting tables?